perception/traffic_light_arbiter/src/traffic_light_arbiter.cpp

```diff
@@ -40,6 +41,26 @@ std::vector<TrafficLightConstPtr> filter_traffic_signals(const LaneletMapConstPt
       signals.push_back(signal);
     }
   }
+
+  return signals;
+}
+
+std::vector<TrafficLightConstPtr> filter_pedestrian_signals(const LaneletMapConstPtr map)
+{
+  namespace query = lanelet::utils::query;
+
+  const auto all_lanelets = query::laneletLayer(map);
+  const auto crosswalks = query::crosswalkLanelets(all_lanelets);
+  std::vector<TrafficLightConstPtr> signals;
+
+  for (const auto & crosswalk : crosswalks) {
+    const auto traffic_light_reg_elems =
+      crosswalk.regulatoryElementsAs<const lanelet::TrafficLight>();
+    std::transform(
```

Comment:
nit: would not `std::copy` suit this better?

Reply:
Addressed in https://github.com/autowarefoundation/autoware.universe/pull/6423/commits/ffaa00d90bba34bdde7a55f00e78f69b1d978715